### PR TITLE
Flex 2971 automate regression test c151 change layout device list sequal

### PR DESCRIPTION
**Pull request by SamVanDijk**

### CODE REVIEW

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/Defaults.java [hunk 1]
```diff
@@ -205,6 +205,7 @@
     public static final String SMSTYPE = "?";
     public static final String LIGHTMEASUREMENT_LASTMESSAGE = "";
     public static final Integer SMS_INDEX = 1;
-    
+
     public static final boolean DEFAULT_BYPASS_RETRY = false;
-}
+    public static final Boolean LAST_KNOWN_STATE = false;
+}
```
Comment (Joep-Simons) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/Defaults.java]: Make sure all files have a newline as last line.

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/Keys.java [hunk 2]
```diff
@@ -285,4 +285,5 @@
     public static final String MESSAGE = "Message";
     public static final String NUMBER_OF_EVENTS = "NumberOfEvents";
     public static final String RANDOM_PLATFORM = "RandomPlatform";
-}
+    public static final String LAST_KNOWN_STATE = "LastKnownState";
+}
```
Comment (Joep-Simons) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/Keys.java]: Make sure all files have a newline as last line.

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/database/CoreDatabase.java [hunk 3]
```diff
@@ -158,4 +163,4 @@ public void prepareDatabaseForScenario() {
 
         this.insertDefaultData();
     }
-}
+}
```
Comment (Joep-Simons) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/database/CoreDatabase.java]: Make sure all files have a newline as last line.

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/database/core/DeviceAuthorizationSteps.java [hunk 4]
```diff
@@ -61,4 +63,4 @@ public void aDeviceAuthorization(final Map<String, String> settings) throws Thro
 
         this.deviceAuthorizationRepository.save(authorization);
     }
-}
+}
```
Comment (Joep-Simons) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/database/core/DeviceAuthorizationSteps.java]: Make sure all files have a newline as last line.

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/database/core/SsldDeviceSteps.java [hunk 5]
```diff
@@ -73,6 +79,27 @@ public Ssld anSsldDevice(final Map<String, String> settings) throws Throwable {
...
+
+        dosList.add(new DeviceOutputSetting(1, 1, RelayType.TARIFF, "test-relay"));
+
+        dosList.add(new DeviceOutputSetting(2, 2, RelayType.LIGHT, "test-relay"));
+
+        ssld.updateOutputSettings(dosList);
+
+        this.relayStatusRepository.save(new RelayStatus(ssld, 1, false, DateTime.now().toDate()));
+
+        this.relayStatusRepository.save(new RelayStatus(ssld, 2, true, DateTime.now().toDate()));
```
Comment (Joep-Simons) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/glue/steps/database/core/SsldDeviceSteps.java]: How does this work when you want to test a device with a different configuration for the relays?

**Comment by CLAAdmin** [comment]
[CLA assistant check](http://cla.opensmartgridplatform.org:80/OSGP/Integration-Tests?pullRequest=209) <br/>All committers of the pull request should sign our Contributor License Agreement in order to get your pull request merged.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: SamVanDijk<br/>:x: web-flow

**Comment by SamVanDijk** [comment]
restest this please

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/Defaults.java [hunk 6]
```diff
@@ -176,6 +176,7 @@
     public static final String DC_LIGHTS = "";
     public static final int FIRMWARE_ID = 0;
     public static final String FIRMWARE_DESCRIPTION = "Test Firmware";
+	public static final Boolean IN_MAINTENANCE = false;
```
Comment (Joep-Simons) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/Defaults.java]: This should be spaces instead of tab.

Comment (Joep-Simons) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/Defaults.java]: Is this a 'boolean' or a 'Boolean'?

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/Defaults.java [hunk 7]
```diff
@@ -207,6 +208,18 @@
     public static final Integer SMS_INDEX = 1;
 
     public static final boolean DEFAULT_BYPASS_RETRY = false;
+    public static final Boolean LAST_KNOWN_STATE = false;
```
Comment (Joep-Simons) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/Defaults.java]: Is this a 'boolean' or a 'Boolean'?

**Comment by SamVanDijk** [comment]
retest this please

**Comment by Joep-Simons** [comment]
retest this please

**Comment by Joep-Simons** [comment]
retest this please

**Pull request merged**